AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthServiceBehavior.swift

```diff
@@ -12,9 +12,67 @@ public protocol AWSAuthServiceBehavior: AnyObject {
 
     func getCredentialsProvider() -> AWSCredentialsProvider
 
+    @available(*, deprecated, message: "Use getIdentityId(completion: (Result<String, AuthError>) -> Void) instead")
     func getIdentityId() -> Result<String, AuthError>
 
+    @available(*, deprecated, message: "Use getToken(completion: (Result<String, AuthError>) -> Void) instead")
     func getToken() -> Result<String, AuthError>
 
     func getTokenClaims(tokenString: String) -> Result<[String: AnyObject], AuthError>
+
+    /// Retrieves the identity identifier of for the Auth service
+    /// - Parameter completion: Completion handler defined for the input `Result<String, AuthError>`
+    func getIdentityId(completion: @escaping (Result<String, AuthError>) -> Void)
+
+    /// Retrieves the token from the Auth token provider
+    /// - Parameter completion: Completion handler defined for the input `Result<String, AuthError>`
+    func getToken(completion: @escaping (Result<String, AuthError>) -> Void)
+}
+
+extension AWSAuthServiceBehavior {
+    // MARK: List of Amplify internal usages of now deprecated AWSAuthServiceBehavior methods.
+    /**
+     `StorageAccessLevelAwarePrefixResolver`
+        - File Path: `AmplifyPlugins/Storage/AWSS3StoragePlugin/Configuration/AWSS3PluginPrefixResolver.swift`
+        - Uses: `getIdentityId()`
+     
+     `IncomingAsyncSubscriptionEventPublisher`
+        - File Path: `AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthServiceBehavior.swift`
+        - Uses: `getToken()`
+     
+     `BasicUserPoolTokenProvider`
+        - File Path: `AmplifyPlugins/Core/AWSPluginsCore/Auth/Provider/AuthTokenProvider.swift`
+        - Uses: `getToken()`
+     
+     `AWSOIDCAuthProvider`
+        - File Path: `AmplifyPlugins/API/AWSAPICategoryPlugin/SubscriptionFactory/AWSOIDCAuthProvider.swift`
+        - Uses: `getToken()`
+     
+     `AWSAuthServiceBehavior` protocol extension. Default implementation of new completion handler APIs to prevent breaking change.
+         - File Path: `AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthServiceBehavior.swift`
+         - Uses: `getToken()`, `getIdentityId()`
+     */
+    
+    
+    /// Retrieves the identity identifier of for the Auth service
+    /// - Parameter completion: Completion handler defined for the input `Result<String, AuthError>`
+    /// - Note: This default implementation was added to prevent a breaking change,
+    /// and will be removed when the blocking API versions are removed.
+    public func getIdentityId(completion: @escaping (Result<String, AuthError>) -> Void) {
+        DispatchQueue.global(qos: .utility).async { [weak self] in
```

Comment:
I would remove `qos: .utility` so that it can inherit priority.

Reply:
Would it inherit priority? I believe it would use the [`default`](https://developer.apple.com/documentation/dispatch/dispatchqos/qosclass/default) QOS ([`global` method doc](https://developer.apple.com/documentation/dispatch/dispatchqueue/2300077-global)).
Do we want to be using `default` QOS here?

Reply:
In agreement after further conversation. Specifically, the insight that these requests generally aren't making network requests.
Addressed in afa08aa.